### ISSUE TITLE
Object store is failing to reconcile in CI against Ceph main images

### ISSUE THREAD

**Issue by travisn** [rook/rook#10631]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
The `smoke-suite-ceph-master` tests of the [daily rook CI](https://github.com/rook/rook/actions/workflows/daily-nightly-jobs.yml) suite is failing to reconcile the object store against the Ceph main image. Setting the application on the `.rgw.root` pool is failing. 
```
2022-07-24 00:49:14.153030 I | ceph-object-controller: reconciling object store deployments
2022-07-24 00:49:14.170367 I | ceph-object-controller: ceph object store gateway service running at 10.100.15.87
2022-07-24 00:49:14.170389 I | ceph-object-controller: reconciling object store pools
2022-07-24 00:49:14.742588 W | ceph-object-controller: failed to adjust the PG count for rgw metadata pools. using the general default. failed to get config setting "rgw_rados_pool_pg_num_min" for user "mon.": exit status 2
2022-07-24 00:49:18.348929 I | cephclient: setting pool property "compression_mode" to "passive" on pool "lite-store.rgw.buckets.non-ec"
2022-07-24 00:49:18.379032 I | cephclient: setting pool property "compression_mode" to "passive" on pool "lite-store.rgw.buckets.index"
2022-07-24 00:49:18.592529 I | cephclient: setting pool property "compression_mode" to "passive" on pool "lite-store.rgw.meta"
2022-07-24 00:49:18.662007 I | cephclient: setting pool property "compression_mode" to "passive" on pool "lite-store.rgw.control"
2022-07-24 00:49:18.795078 I | cephclient: setting pool property "compression_mode" to "passive" on pool "lite-store.rgw.log"
2022-07-24 00:49:20.323296 I | cephclient: setting pool property "compression_mode" to "passive" on pool ".rgw.root"
2022-07-24 00:49:21.807959 I | cephclient: application "rook-ceph-rgw" is already set on pool "lite-store.rgw.buckets.non-ec"
2022-07-24 00:49:21.808165 I | cephclient: reconciling replicated pool lite-store.rgw.buckets.non-ec succeeded
2022-07-24 00:49:22.227909 E | cephclient: failed to set property "compression_mode" to pool ".rgw.root" to "passive". failed to set pool property "compression_mode" on pool ".rgw.root": exit status 2
2022-07-24 00:49:22.284689 I | cephclient: application "rook-ceph-rgw" is already set on pool "lite-store.rgw.buckets.index"
2022-07-24 00:49:22.284708 I | cephclient: reconciling replicated pool lite-store.rgw.buckets.index succeeded
2022-07-24 00:49:22.972391 I | cephclient: application "rook-ceph-rgw" is already set on pool "lite-store.rgw.meta"
2022-07-24 00:49:22.972411 I | cephclient: reconciling replicated pool lite-store.rgw.meta succeeded
2022-07-24 00:49:23.226133 I | cephclient: application "rook-ceph-rgw" is already set on pool "lite-store.rgw.control"
2022-07-24 00:49:23.226154 I | cephclient: reconciling replicated pool lite-store.rgw.control succeeded
2022-07-24 00:49:23.419930 I | cephclient: application "rook-ceph-rgw" is already set on pool "lite-store.rgw.log"
2022-07-24 00:49:23.420129 I | cephclient: reconciling replicated pool lite-store.rgw.log succeeded
2022-07-24 00:49:23.505680 E | ceph-object-controller: failed to reconcile CephObjectStore "smoke-ns/lite-store". failed to create object store deployments: failed to create object pools: failed to create metadata pools: failed to create pool ".rgw.root": failed to tag pool ".rgw.root" for application "rook-ceph-rgw": failed to get application for pool ".rgw.root": failed to get current application for pool .rgw.root: exit status 2
```

**Expected behavior:**
Daily tests against Ceph main should pass.

**How to reproduce it (minimal and precise):**

Configure the rook object store against the ceph main image: `quay.io/ceph/daemon-base:latest-master-devel`

**Comment by travisn** [comment]
Oh I should have remembered that the issue is that Rook still can't detect the difference for Reef builds due to https://github.com/ceph/ceph/pull/45817. We need that to merge to get this fixed, so you can close #10635 as not necessary, thanks.

**Comment by travisn** [comment]
The CI is passing now against the latest reef images from ceph main
